compiler/rustc_borrowck/src/type_check/relate_tys.rs

```diff
@@ -84,7 +84,7 @@ impl<'me, 'bccx, 'tcx> NllTypeRelatingDelegate<'me, 'bccx, 'tcx> {
     }
 }
 
-impl<'tcx> TypeRelatingDelegate<'tcx> for NllTypeRelatingDelegate<'_, '_, 'tcx> {
+impl<'tcx> NllTypeRelatingDelegate<'tcx> for NllNllTypeRelatingDelegate<'_, '_, 'tcx> {
```

Comment:
this is now the only `NllNllTypeRelatingDelegate` in existance, just stop using a delegate and inline this into the relation? That does require moving the whole relation to `rustc_borrowck` which... does that cause some privacy issues?

Also... one `Nll` to many here

i am in generally in favor of this change

Reply:
NllNll was a typo, lol

I just inlined the whole thing into borrowck